### PR TITLE
Show permissions on Snap details page

### PR DESCRIPTION
**Pull request by Mrtenz**
This adds permissions to the Snap details page.

### CODE REVIEW

File: src/features/snap/permissions.tsx [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+/**
+ * Get the label to use for a network.
+ *
+ * @param networkId - The ID of the network to get the name for.
+ * @param fallback - The fallback name to use if the network is unknown.
+ * @returns The label for the network, or the fallback if the network is
+ * unknown.
+ */
+function getPublicKeyNetworkLabel(
```
Comment (FrederikBolding) [inline src/features/snap/permissions.tsx:104]: Could we use these functions from snaps-utils instead of referencing slip44 directly? https://github.com/MetaMask/snaps/blob/479654cb5dbc1a99f7744e20c0950c424ef73fdf/packages/snaps-utils/src/derivation-paths.ts#L157-L199

File: src/features/snap/components/Permissions.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  snap,
+  permissions,
+}) => {
+  const i18n = useLingui();
+  const descriptors = useMemo(
+    () => getPermissions(snap, permissions),
+    [snap, permissions],
+  );
+
+  return (
```
Comment (FrederikBolding) [inline src/features/snap/components/Permissions.tsx:35]: The ticket describes that the permission list should be collapsed if it contains too many permissions to show at once. This does not seem to be in this implementation 🤔

File: src/features/snap/permissions.tsx [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+  | 'audits'
+  | 'banner'
+  | 'support'
+  | 'permissions'
+>;
+
+// Note that we have to define the `PermissionsKey` type separately, since
+// TypeScript doesn't enforce that the keys are present otherwise.
+type PermissionsKey =
+  | Exclude<keyof InitialPermissions, 'snap_confirm'>
```
Comment (FrederikBolding) [inline src/features/snap/permissions.tsx:51]: Why is this required?

File: src/features/snap/permissions.tsx [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+        icon: ExploreIcon,
+        weight: 4,
+      });
+    }
+
+    return result;
+  },
+  'endowment:webassembly': ({ name }) => ({
+    label: defineMessage`Support for WebAssembly`,
+    description: defineMessage`Allow ${name} to access low-level execution environments via WebAssembly.`,
```
Comment (FrederikBolding) [inline src/features/snap/permissions.tsx:218]: Do all of these match the extension implementation? Are they meant to?

Comment (Mrtenz) [reply src/features/snap/permissions.tsx:218]: I copied everything from the extension yes. Not sure what else we should use?

File: src/features/snap/permissions.tsx (hunk 3 above)
Comment (Mrtenz) [reply src/features/snap/permissions.tsx:51]: `InitialPermissions` does not include these permissions, and `snap_confirm` was removed a long time ago. We should add the missing ones 😅

File: src/features/snap/components/Permissions.tsx (hunk 2 above)
Comment (Mrtenz) [reply src/features/snap/components/Permissions.tsx:35]: I'm not sure what the best way to do that is, and it doesn't really seem to be a problem yet. Possibly something we can do later.

Comment (FrederikBolding) [reply src/features/snap/components/Permissions.tsx:35]: For some snaps it's a lot already, but I want to hear what @eriknson thinks

image

File: src/features/snap/permissions.tsx (hunk 3 above)
Comment (FrederikBolding) [reply src/features/snap/permissions.tsx:51]: Does `InitialPermissions` contain `snap_confirm` still? 😭

File: src/features/snap/permissions.tsx (hunk 4 above)
Comment (FrederikBolding) [reply src/features/snap/permissions.tsx:218]: I think they should match, I was just asking a clarifying question 👍

File: src/features/snap/permissions.tsx (hunk 1 above)
Comment (Mrtenz) [reply src/features/snap/permissions.tsx:104]: Using `snaps-utils` results in errors like this:

```
ERROR in ./node_modules/@metamask/snaps-utils/dist/esm/virtual-file/toVirtualFile.js 1:0-44
Module not found: Error: Can't resolve 'fs' in './node_modules/@metamask/snaps-utils/dist/esm/virtual-file'
```

I don't think we should include polyfills for these things since it's not used. 😕

Comment (FrederikBolding) [reply src/features/snap/permissions.tsx:104]: Huh, is it not using the browser build? 🤔 

https://github.com/MetaMask/snaps/blob/main/packages/snaps-utils/src/virtual-file/index.browser.ts

File: src/features/snap/permissions.tsx (hunk 3 above)
Comment (Mrtenz) [reply src/features/snap/permissions.tsx:51]: Fixed in MetaMask/snaps#2111. Will update once that's released.

**Review by FrederikBolding — APPROVED** [review]
